src/Graph/SubGraph.ts

```diff
@@ -0,0 +1,289 @@
+    const mosaicConfig: MosaicConfig = MosaicConfig.fromChain(this.originChain);
+    return {
+      projectRoot: Directory.projectRoot,
+      ostComposerAddress: mosaicConfig.originChain.contractAddresses.ostComposerAddress,
+      eip20GatewayAddress: mosaicConfig.auxiliaryChains[this.auxiliaryChain].contractAddresses.origin.ostEIP20GatewayAddress,
+    };
+  }
+
+  /**
+   * Returns values for all template variables which need to be replaced in subgraph.yaml for auxiliary subGraphType
+   */
+  private auxiliaryChainTemplateVariables(): object {
+    const mosaicConfig: MosaicConfig = MosaicConfig.fromChain(this.originChain);
+    const auxiliaryContractAddresses = mosaicConfig.auxiliaryChains[this.auxiliaryChain].contractAddresses.auxiliary;
+    return {
+      projectRoot: Directory.projectRoot,
+      anchorAddress: auxiliaryContractAddresses.anchorAddress,
+      eip20CoGatewayAddress: auxiliaryContractAddresses.ostEIP20CogatewayAddress,
+    };
+  }
+
+  /**
+   * Create local instance and deploy sub graph.
+   */
+  private deployLocal(): {success: boolean, message: string} {
+    this.logInfo('attempting to deploy local graph');
+    try {
+      this.executeGraphCommand(
+        `deploy --node http://localhost:${this.graphDescription.rpcAdminPort}/ --ipfs http://localhost:${this.graphDescription.ipfsPort} ${this.name}`,
+      );
+      return{ success: true, message: '' };
+    } catch (ex) {
+      const message = this.extractMessageFromError(ex);
+      this.logInfo(`deploy local graph failed with: ${message}`);
+      this.logInfo('removing local graph');
+      this.executeGraphCommand(`remove --node http://localhost:${this.graphDescription.rpcAdminPort}/ ${this.name}`);
+      return { success: false, message };
+    }
+  }
+
+  /**
+   * Execute graph command.
+   * @param {string} commandSuffix
+   */
+  private executeGraphCommand(commandSuffix: string): void {
+    Shell.executeInShell(`cd ${this.getTempGraphInstallationDir} && ./node_modules/.bin/graph ${commandSuffix}`);
+  }
+
+  /**
+   * Create local instance and deploy graph.
+   */
+  private deleteCodeFromTempDir(): void {
+    this.logInfo('deleting auto generated code from temp directory');
+    FileSystem.removeSync(this.getTempGraphInstallationDir);
+  }
+
+  /**
+   * Persist auto generated code.
+   */
+  private copyToSubGraphProjectDir(): void {
+    this.logInfo('persisting auto generated graph code');
+    const subGraphProjectDir = this.getSubGraphProjectDir;
+    FileSystem.ensureDirSync(subGraphProjectDir);
+    FileSystem.copySync(
+      this.getTempGraphInstallationDir,
+      subGraphProjectDir,
+    );
+  }
+
+  /**
+   * Extract message from error object.
+   * @param {Error} error
+   * @return {string}
+   */
+  private extractMessageFromError(error: Error): string {
+    const jsonError = JSON.stringify(error);
+    if (jsonError['error']) {
+      return jsonError['error'];
+    } else if (jsonError['stderr']) {
```

Comment:
If the `if` returns, `else` is not required. Below is more readable. 

```
    if (jsonError['error']) {
      return jsonError['error'];
    }
    if (jsonError['stderr']) {
      return jsonError['stderr'].trim.toString();
    }
    if (jsonError['stdout']) {
      return jsonError['stdout'].trim.toString();
    }
     return 'Something went wrong';
```

Reply:
Agree. Will keep this in mind going forward.